src/request_builder.rs

```diff
@@ -0,0 +1,377 @@
+    use crate::{identifiers::RoomId, AsyncClient, Session};
+
+    use js_int::Int;
+    use mockito::{mock, Matcher};
+    use std::convert::TryFrom;
+    use url::Url;
+
+    #[tokio::test]
+    async fn create_room_builder() {
+        let homeserver = Url::parse(&mockito::server_url()).unwrap();
+
+        let _m = mock("POST", "/_matrix/client/r0/createRoom")
+            .with_status(200)
+            .with_body_from_file("./tests/data/room_id.json")
+            .create();
+
+        let session = Session {
+            access_token: "1234".to_owned(),
+            user_id: UserId::try_from("@example:localhost").unwrap(),
+            device_id: "DEVICEID".to_owned(),
+        };
+
+        let mut bldr = RoomBuilder::new();
+        bldr.creation_content(false)
+            .initial_state(vec![])
+            .visibility(Visibility::Public)
+            .name("room_name")
+            .room_version("v1.0")
+            .invite_3pid(vec![])
+            .is_direct(true)
+            .power_level_override(PowerLevelsEventContent {
+                ban: Int::max_value(),
+                events: HashMap::default(),
+                events_default: Int::min_value(),
+                invite: Int::min_value(),
+                kick: Int::min_value(),
+                redact: Int::max_value(),
+                state_default: Int::min_value(),
+                users_default: Int::min_value(),
+                notifications: NotificationPowerLevels {
+                    room: Int::min_value(),
+                },
+                users: HashMap::default(),
+            })
+            .preset(RoomPreset::PrivateChat)
+            .room_alias_name("room_alias")
+            .topic("room topic")
+            .visibility(Visibility::Private);
+        let mut cli = AsyncClient::new(homeserver, Some(session)).unwrap();
+        assert!(cli.create_room(bldr).await.is_ok());
+    }
+
+    #[tokio::test]
+    async fn get_message_events() {
+        let homeserver = Url::parse(&mockito::server_url()).unwrap();
+
+        let _m = mock(
+            "GET",
+            Matcher::Regex(r"^/_matrix/client/r0/rooms/.*/messages".to_string()),
+        )
+        .with_status(200)
+        .with_body_from_file("./tests/data/room_messages.json")
+        .create();
+
+        let session = Session {
+            access_token: "1234".to_owned(),
+            user_id: UserId::try_from("@example:localhost").unwrap(),
+            device_id: "DEVICEID".to_owned(),
+        };
+
+        let mut bldr = RoomMessageBuilder::new();
+        bldr.room_id(RoomId::try_from("!roomid:example.com").unwrap())
+            .from("t47429-4392820_219380_26003_2265".to_string())
+            .to("t4357353_219380_26003_2265".to_string())
+            .direction(Direction::Backward)
+            .limit(UInt::new(10).unwrap());
+        // TODO this makes ruma error `Err(IntoHttp(IntoHttpError(Query(Custom("unsupported value")))))`??
```

Comment:
Sounds like `RoomEventFilter::default()` can't be converted into a query string, perhaps the `default()` has some value that can't be converted into a query string or it isn't yet implemented properly in ruma.

Reply:
Should I ask in the Ruma room or file an issue maybe?

Reply:
Sure, perhaps someone knows what's going on.

Reply:
I opened a PR [for this](https://github.com/ruma/ruma-client-api/pull/141)